addons/hr_holidays/models/hr_leave.py

```diff
@@ -797,9 +847,28 @@ def onchange(self, values, field_name, field_onchange):
         # Try to force the leave_type name_get when creating new records
         # This is called right after pressing create and returns the name_get for
         # most fields in the view.
+        context = {}
         if field_onchange.get('employee_id') and 'employee_id' not in self._context and values:
             employee_id = get_employee_from_context(values, self._context, self.env.user.employee_id.id)
-            self = self.with_context(employee_id=employee_id)
+            context['employee_id'] = employee_id
+            context['default_employee_id'] = employee_id
+        elif not values:
+            context['employee_id'] = self.env.user.employee_id.id
+            context['default_employee_id'] = self.env.user.employee_id.id
+
+        if values.get('id') and values.get('request_date_from') and values.get('request_date_to'):
+            context['default_date_from'] = values.get('request_date_from'),
+            context['default_date_to'] = values.get('request_date_to')
```

Comment:
As you know that the values are there, you don't need to use `get()`
```suggestion
            context['default_date_from'] = values['request_date_from']
            context['default_date_to'] = values['request_date_to']
```